ddtrace/internal/processor/endpoint_call_counter.py

```diff
@@ -0,0 +1,44 @@
+import typing
+
+import attr
+
+from ddtrace.ext import SpanTypes
+from ddtrace.internal import nogevent
+from ddtrace.internal.compat import ensure_str
+from ddtrace.internal.processor import SpanProcessor
+from ddtrace.span import Span
+
+
+EndpointCountsType = typing.Dict[str, int]
+
+
+@attr.s(eq=False)
+class EndpointCallCounterProcessor(SpanProcessor):
+
+    endpoint_counts = attr.ib(init=False, repr=False, type=EndpointCountsType, factory=lambda: {}, eq=False)
+    _endpoint_counts_lock = attr.ib(init=False, repr=False, factory=nogevent.DoubleLock, eq=False)
+    _enabled = attr.ib(default=False, repr=False, eq=False)
+
+    def enable(self):
+        # type: () -> None
+        self._enabled = True
+
+    def on_span_start(self, span):
+        # type: (Span) -> None
+        pass
+
+    def on_span_finish(self, span):
+        # type: (Span) -> None
+        if not self._enabled:
+            return
+        if span._local_root == span and span.span_type == SpanTypes.WEB and span.resource is not None:
```

Comment:
Are `SpanTypes.WEB` not always local roots?

Reply:
I'm guessing that it would be possible to make 'manual' spans that are not. Also I'm not sure we are 'guaranteed' for them to be local roots - IIRC we've seen examples in the past when they are not.

Reply:
@mar-kolya verify that this logic matches what we do in profiler when collect CPU samples

Reply:
So we do this here:

https://github.com/DataDog/dd-trace-py/blob/747c03653f16445aabde9c53d66159cb630bde60/ddtrace/profiling/exporter/pprof.pyx#LL665C10-L665C85

(check that this web span)

and here:

https://github.com/DataDog/dd-trace-py/blob/e8681f7fdf28849885686444eee03e9537160482/ddtrace/profiling/event.py#L72

- we check that this is local root span.